flyway-core/src/main/java/org/flywaydb/core/internal/dbsupport/oracle/OracleSchema.java

```diff
@@ -339,39 +353,86 @@ private boolean spatialExtensionsAvailable() throws SQLException {
+                        "      JOIN all_part_tables apt " +
+                        "        ON apt.table_name = at.table_name AND " +
+                        "           apt.owner = at.owner AND " +
+                        "           apt.ref_ptn_constraint_name IS NOT NULL " +
+                        "      JOIN all_constraints c1 " +
+                        "        ON apt.owner = c1.owner AND " +
+                        "           apt.ref_ptn_constraint_name = c1.constraint_name AND " +
+                        "           c1.constraint_type = 'R' " +
+                        "      JOIN all_constraints c2 " +
+                        "        ON c2.owner = c1.owner AND " +
+                        "           c2.constraint_name = c1.r_constraint_name AND " +
+                        "           c2.constraint_type = 'P' " +
+                        "), " +
+                        // create view of all tables. Referenced partitioned tables are ordered over an hierarchical view (all levels)
+                        "  all_table_levels(table_name, owner, parent_table_name, parent_owner, lev) AS ( " +
+                        "      SELECT at.table_name, at.owner, NULL, NULL, 1 AS lev " +
+                        "      FROM all_tables at " +
+                        "        LEFT JOIN all_part_tables apt " +
+                        "          ON apt.table_name = at.table_name AND " +
+                        "             apt.owner = at.owner " +
+                        "      WHERE at.owner = ? AND " +
+                        "                        at.table_name NOT LIKE 'BIN$%' AND " +
+                        "                        at.table_name NOT LIKE 'MDRT_%$' AND " +
+                        "                          at.table_name NOT LIKE 'MLOG$%' AND " +
+                        "                                            at.table_name NOT LIKE 'RUPD$%' AND " +
+                        "      at.table_name NOT LIKE 'DR$%' AND " +
+                        "      at.table_name NOT LIKE 'SYS_IOT_OVER_%' AND " +
+                        "                    at.nested != 'YES' AND " +
+                        "      at.secondary != 'Y' AND " +
+                        "      apt.ref_ptn_constraint_name IS NULL " +
+                        "                                     UNION ALL " +
+                        "                                     SELECT h.table_name, h.owner, h.parent_table_name, h.parent_owner, r.lev + 1 AS lev " +
+                        "                                                                                                        FROM ref_part_dependencies h " +
+                        "      JOIN all_table_levels r " +
+                        "      ON h.parent_table_name = r.table_name AND " +
+                        "      h.parent_owner = r.owner " +
+                        "  ) " +
+                        "SELECT table_name FROM all_table_levels ORDER BY lev DESC", name);
```

Comment:
Overcomplicated. Should be:
all_tables LEFT JOIN all_part_tables LEFT  JOIN all_constraints LEFT JOIN all_constraints + CONNECT BY from parents to children starting with parent is null